packages/x-charts/src/internals/domUtils.ts

```diff
@@ -50,36 +54,28 @@ function autoCompleteStyle(name: string, value: number) {
  * @param text camelcase css property
  * @returns css property
  */
-function camelToMiddleLine(text: string) {
-  const strs = text.split('');
-
-  const formatStrs = strs.reduce((result: string[], entry) => {
-    if (entry === entry.toUpperCase()) {
-      return [...result, '-', entry.toLowerCase()];
-    }
-
-    return [...result, entry];
-  }, []);
-
-  return formatStrs.join('');
```

Comment:
Regex should be faster than creating an array of chars and re-allocating an array on each reduce step.